Adding a new VMware utilities module

jcpowermac opened the pull request:
This PR adds the support for VMware utility module for frequently used functions such as:

`wait_for_task`
`find_datacenter_by_name`
`vmware_argument_spec`
`connect_to_api`
`get_all_objs`

This [PR](https://github.com/ansible/ansible-modules-core/pull/1154) for the cluster and datacenter is using this module and has been tested.

jcpowermac committed:
Adding a new VMware utilities module

bcoca commented on lib/ansible/module_utils/vmware.py:
```diff
@@ -0,0 +1,181 @@
+#!/bin/python
```
this is for import, should not have a shebang

bcoca commented on lib/ansible/module_utils/vmware.py:
```diff
@@ -0,0 +1,181 @@
+        return None
+    # This exception should be handled in the module that calls this method
+    # and fail with an appropriate message to module.fail_json()
+    except vmodl.MethodFault:
```
useless try/except, several of these in the code

bcoca commented on lib/ansible/module_utils/vmware.py:
```diff
@@ -0,0 +1,181 @@
+    except vim.fault.InvalidLogin as invalid_login:
+        module.fail_json(msg=invalid_login.msg)
+    except requests.ConnectionError:
+        module.fail_json(msg="Unable to connect to vCenter or ESXi API on TCP/443.")
```
it is useful to also add the api exception info:
module.fail_json(msg="Unable to connect to vCenter or ESXi API on TCP/443.", apierror=str(e))

bcoca commented on lib/ansible/module_utils/vmware.py:
```diff
@@ -0,0 +1,181 @@
+    return False
+
+
+def task_error(task):
```
I'm not sure this function is useful

bcoca commented on lib/ansible/module_utils/vmware.py:
```diff
@@ -0,0 +1,181 @@
+
+
+def task_running(task):
+    time.sleep(15)
```
task doesn't seem to be used here

bcoca commented on lib/ansible/module_utils/vmware.py:
```diff
@@ -0,0 +1,181 @@
+
+
+def task_success(task):
+    return True
```
not sure if this is useful

jcpowermac committed:
Modified per @bcoca
Removed try/except raises
Modified wait_for_task
Added api exception error message

jcpowermac commented:
The replacement module-extras [PR](https://github.com/ansible/ansible-modules-extras/pull/415) specifically for adding datacenter objects within vCenter, which uses this utility module.

bcoca commented on lib/ansible/module_utils/vmware.py:
```diff
@@ -0,0 +1,137 @@
+    return None
+
+
+def find_datacenter_by_name(content, datacenter_name, throw=True):
```
throw seems unused
